smartfile/__init__.py

```diff
@@ -124,8 +129,32 @@ def put(self, endpoint, id=None, **kwargs):
     def post(self, endpoint, id=None, **kwargs):
         return self._request('post', endpoint, id=id, data=kwargs)
 
-    def delete(self, endpoint, id=None, **kwargs):
-        return self._request('delete', endpoint, id=id, data=kwargs)
+    def delete(self, deletefile):
+        return self.post('/path/oper/remove', path=deletefile)
+
+    def upload(self, filename, fileobj):
+        if filename.endswith('/'):
+            raise ValueError("File name should have no trailing slash")
+        arg = (filename, fileobj)
+        return self.post('/path/data/', file=arg)
+
+    def download(self, file_to_be_downloaded):
+        """ file_to_be_downloaded is a file-like object that has already
+        been uploaded, you cannot download folders """
+        # download uses shutil.copyfileobj to download, which copies
+        # the data in chunks
+        o = open(file_to_be_downloaded, 'wb')
+        return shutil.copyfileobj(self.get('/path/data/',
+                                  file_to_be_downloaded), o)
+
+    def move(self, src_path, dst_path):
+        # check destination folder for / at end
+        if not dst_path.endswith("/"):
+            dst_path = dst_path + "/"
+        # check destination folder for / at begining
+        if not dst_path.startswith("/"):
+            dst_path = "/" + dst_path
```

Comment:
aren't you supposed to do the same also for `src_path` (prepend and append slashes)?